platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -334,6 +334,30 @@ def get_name(self):
     def get_description(self):
         return self.description
 
+    def auto_update_firmware(self, image_path, boot_action):
+        """
+        Default handling of attempted automatic update for a component of a Mellanox switch.
+        Will skip the installation if the boot_action is 'warm' or 'fast' and will call update_firmware()
+        if boot_action is fast.
+        """
+
+        default_supported_boot = ['cold']
+
+        # Verify image path exists
+        if not os.path.exists(image_path):
+            # Invalid image path
+            return -2
```

Comment:
please avoid using magic numbers, suggest defining some more readable constants.  same comments apply to the below code changes.

Reply:
Agreed, although I think the most appropriate location for these definitions would be sonic_platform_base as these numbers are "defined" by the API. Should I attempt to add them to there?